emscripten.py

```diff
@@ -452,8 +338,202 @@ def move_preasm(m):
       logging.debug('  emscript: python processing: function tables and exports took %s seconds' % (time.time() - t))
 
     return (post, funcs_js, sending, receiving, asm_setup, the_global, asm_global_vars,
-            asm_global_funcs, pre_tables, final_function_tables, exports,
-            function_table_data, forwarded_json)
+            asm_global_funcs, pre_tables, final_function_tables, exports, function_table_data)
+
+
+def finalize_funcs_js(base_funcs_js, asm_setup, the_global, sending, receiving, asm_global_vars,
+                      asm_global_funcs, pre_tables, final_function_tables,
+                      exports, metadata, settings):
+  receiving += create_named_globals(metadata, settings)
+  runtime_funcs = create_runtime_funcs(exports, settings)
+
+  asm_start_pre = create_asm_start_pre(asm_setup, the_global, sending, metadata, settings)
+  asm_temp_vars = create_asm_temp_vars(settings)
+  asm_start = asm_start_pre + '\n' + asm_global_vars + asm_temp_vars + '\n' + asm_global_funcs
+
+  temp_float = '  var tempFloat = %s;\n' % ('Math_fround(0)' if provide_fround(settings) else '0.0')
+  async_state = '  var asyncState = 0;\n' if settings.get('EMTERPRETIFY_ASYNC') else ''
+  f0_fround = '  const f0 = Math_fround(0);\n' if provide_fround(settings) else ''
+
+  replace_memory = create_replace_memory(settings)
+
+  start_funcs_marker = '\n// EMSCRIPTEN_START_FUNCS\n'
+
+  asm_end = create_asm_end(exports, settings)
+
+  runtime_library_overrides = create_runtime_library_overrides(settings)
+
+  funcs_js = [
+    asm_start,
+    temp_float,
+    async_state,
+    f0_fround,
+    replace_memory,
+    start_funcs_marker
+  ] + runtime_funcs + base_funcs_js + ['\n  ',
+    pre_tables, final_function_tables, asm_end,
+    '\n', receiving, ';\n', runtime_library_overrides]
+
+  return funcs_js
+
+
+def finalize_output(metadata, post, funcs_js, function_table_data, settings, outfile, DEBUG):
```

Comment:
Shuffled internals and renamed to hopefully be more clear.

Avoiding docstrings for the moment because ideally docstrings would be *prescriptive*, whereas I've been discovering these functions as I go through them, so I only know them from a *descriptive* standpoint, and that's probably less useful than no docstring.